opsdroid/constraints.py

```diff
@@ -23,13 +24,16 @@ def constraint_callback(message, rooms=rooms):
             return message.target in rooms
 
         func = add_skill_attributes(func)
-        func.constraints.append(constraint_callback)
+        if not invert:
```

Comment:
I would prefer if this was not a negative. You could always reassign the function using your negate decorator. For example:

```
if invert:
    constraint_callback = negate(constraint_callback)
```

Then append the contraint callback as before.

Same for all other instances of this pattern.